raft/log.hh

```diff
@@ -65,9 +67,10 @@ private:
     void init_last_conf_idx();
     log_entry_ptr& get_entry(index_t);
     const log_entry_ptr& get_entry(index_t) const;
+    size_t range_memory_usage(log_entries::iterator first, log_entries::iterator last) const;
 public:
-    explicit log(snapshot_descriptor snp, log_entries log = {})
-            : _snapshot(std::move(snp)), _log(std::move(log)) {
+    explicit log(snapshot_descriptor snp, log_entries log = {}, size_t max_command_size = sizeof(log_entry))
```

Comment:
> I don't like default values where there is no obvious choice for the default value, and I don't think there is an obvious choice in this case. Well, I generally don't like them because it's easy to forget about some parameter and introduce a bug, the compiler not complaining.

I had this particular argument in mind, and generally I agree with it, but it would mean to generate a lot of changes in `fsm_test.cc`, we would need to create a helper `create_test_log()` to not clutter the tests, etc. It didn't seem worth it.